### PR TITLE
Remove hideTempErr to allow downstream users to check for errors like net.ErrClosed

### PR DESCRIPTION
**Pull request by UnAfraid**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Go Version Update
- [ ] Dependency Update

## Description
Since this change https://github.com/gorilla/websocket/pull/840/files#diff-4f427d2b022907c552328e63f137561f6de92396d7a6e8f6c2ea1bcf0db52654L190-R197 we can no longer determinate if the errors coming from ReadMessage() are net.ErrClosed for example
Hardcoding the error message is not great option because it may vary from OS to OS and system locale

## Related Tickets & Documents
- Related Issue # https://github.com/gorilla/websocket/issues/877
- Closes # https://github.com/gorilla/websocket/issues/877

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Run verifications and test

- [X] `make verify` is passing
- [X] `make test` is passing

### CODE REVIEW

**Comment by GreenMarmot** [comment]
Thank you! The maintainer's decision to hide all errors from the application was mystifying and is one of the reasons we have not upgraded past 1.5.0.

**Comment by StevenACoffman** [comment]
@AlexVulaj This would fix several issues with the 1.5.1 release. Please take a look. Thanks!

File: conn_test.go
Comment (jaitaiwan) [inline conn_test.go:809]: I'm not sure if we really need this test to be run in Parallel?

**Review by jaitaiwan — APPROVED** [review]
LGTM, one small nit. I am a little cautious about this change because I haven't looked at the context of c.readErr. If other Members are happy then I don't see any structural issues with this.

**Comment by GreenMarmot** [comment]
Prior to commit 666c197fc9157896b57515c3a3326c3f8c8319fe, hideTempError wrapped errors with e.Temporary() == true.  All other errors were passed through as is.  The PR that added hideTempError says "Do not return net.Error with Temporary() == true".   This was probably done because errors on the connection are not recoverable.
 
The net.Error Temporary() method in the standard library was recently deprecated because the meaning of the method was not well-defined. 

This PR brings the package closer to 1.50 by not wrapping any errors.

Regarding the parallel test:  #872 sprinkled t.Parallel() through almost all tests, including [tests where it makes no sense](https://github.com/gorilla/websocket/blob/d08ee1ad9b8a1bad1b388165f85f7794a604b8e1/example_test.go#L45).  For better or worse, the call to t.Parallel() goes with the flow.

File: conn_test.go
Comment (UnAfraid) [reply conn_test.go]: Removed t.Parallel

Comment (AlexVulaj) [reply conn_test.go]: I could go either way but I'm in favor of leaving the parallel test for consistency since https://github.com/gorilla/websocket/pull/872 was recently merged.

**Comment by AlexVulaj** [comment]
I'm happy to merge this @jaitaiwan , feel free to make a final call on the "t.Parallel" comment but I don't have any strong feelings.

**Comment by UnAfraid** [comment]
Please let me know if t.Parallel() should be used and i can bring it back, i don't mind either way.

File: conn_test.go
Comment (jaitaiwan) [reply conn_test.go]: Lets leave it as is and not require more changes.

**Review by jaitaiwan — APPROVED** [review]
LGTM

**Comment by StevenACoffman** [comment]
Test and lint failures are unrelated to this change.

**Comment by jaitaiwan** [comment]
Agreed. Thanks for the contribution!

**Pull request merged**
